File: src/main/java/com/bdev/hogwarts_api/connection/SshConnection.java
package com.bdev.hogwarts_api.connection;

import com.bdev.hogwarts_api.properties.PropertiesConfig;
import com.jcraft.jsch.JSch;
import com.jcraft.jsch.Session;
import lombok.SneakyThrows;

import java.util.Properties;

import static com.bdev.hogwarts_api.properties.PropertiesConfig.config;

public class SshConnection {
    private Session session;

    @SneakyThrows
    public SshConnection() {
        JSch jsch = new JSch();

        String host = config.getDatabaseRemoteConnectionHost();
        String login = config.getDatabaseRemoteConnectionUsername();
        String password = config.getDatabaseRemoteConnectionPassword();
        int port = (int) config.getDatabaseRemoteConnectionPort();
        int databaseLocalPort = (int) config.getDatabaseRemoteConnectionLocalPort();
        int databaseRemotePort = (int) config.getDatabaseRemoteConnectionRemotePort();

        session = jsch.getSession(login, host, port); {
            Properties config = new Properties(); {
                config.put("StrictHostKeyChecking", "no");
            }

            session.setPassword(password);
            session.setConfig(config);
            session.connect();
        }

        session.setPortForwardingL(databaseLocalPort, host, databaseRemotePort);
    }

    public void close() {
        session.disconnect();
    }
}

File: src/main/java/com/bdev/hogwarts_api/connection/SshConnection.kt
package com.bdev.hogwarts_api.connection

import com.bdev.hogwarts_api.properties.PropertiesConfig.config
import com.jcraft.jsch.JSch
import com.jcraft.jsch.Session
import java.util.*

class SshConnection {
    private val session: Session

    init {
        val jsch = JSch()

        val host = config.databaseRemoteConnectionHost
        val login = config.databaseRemoteConnectionUsername
        val password = config.databaseRemoteConnectionPassword
        val port = config.databaseRemoteConnectionPort.toInt()
        val databaseLocalPort = config.databaseRemoteConnectionLocalPort.toInt()
        val databaseRemotePort = config.databaseRemoteConnectionRemotePort.toInt()

        session = jsch.getSession(login, host, port)

        session.setPassword(password)
        session.setConfig(getConfig())
        session.setPortForwardingL(databaseLocalPort, host, databaseRemotePort)

        session.connect()
    }

    fun close() {
        session.disconnect()
    }

    private fun getConfig(): Properties {
        val config = Properties()

        config["StrictHostKeyChecking"] = "no"

        return config
    }
}